src/lib/storage/segment_accessor.cpp

```diff
@@ -12,8 +12,29 @@ std::unique_ptr<AbstractSegmentAccessor<T>> CreateSegmentAccessor<T>::create(
   resolve_segment_type<T>(*segment, [&](const auto& typed_segment) {
     using SegmentType = std::decay_t<decltype(typed_segment)>;
     if constexpr (std::is_same_v<SegmentType, ReferenceSegment>) {
-      if (typed_segment.pos_list()->references_single_chunk() && typed_segment.pos_list()->size() > 0) {
-        accessor = std::make_unique<SingleChunkReferenceSegmentAccessor<T>>(typed_segment);
+      const auto& pos_list = *typed_segment.pos_list();
+      if (pos_list.references_single_chunk() && pos_list.size() > 0) {
+        // If the first chunk_id references a non-existing chunk, the entry is NULL. Since all entries reference
```

Comment:
I found the comment a bit confusing in the beginning since the first chunk_id referencing a non-existing chunk is nowhere checked in the code. Simply reversing it would clear it up for me: `If the first entry is NULL, a non-existing chunk is referenced. Since all...` Also, beginners/people not familiar with Hyrise might wonder why the comment mentions `std::nullopt` while the code uses the `NullAccessor`.

Reply:
```
        // If the pos list stores a NULL value, its chunk_id references a non-existing chunk. If all entries reference
        // the same chunk_id, we can safely assume that all other entries are also NULL. Instead of using an accessor
        // that checks for the reference being NULL, we can simply use the NullAccessor, which always returns nullopt,
        // i.e., the accessors representation of NULL values.
        // Note that this is independent of the row being pointed to holding a NULL value.

```
?

Reply:
Perfect